tests/graph_test.py

```diff
@@ -28,13 +30,20 @@ def test_map_raises_when_mapping_over_empty(params) -> None:
         pd.DataFrame({'a': [1, 2], 'c': [1, 2]}),
     ],
 )
-def test_map_raises_if_mapping_nonexistent_node(params) -> None:
+def test_map_adds_node_when_mapping_nonexistent_node(params) -> None:
     g = nx.DiGraph()
     g.add_edge('a', 'b')
 
     graph = cb.Graph(g)
-    with pytest.raises(ValueError):
-        graph.map(params)
+    mapped = graph.map(params)
+    result = mapped.to_networkx()
+    c_data = [
+        data
+        for node, data in result.nodes(data=True)
+        if getattr(node, 'name', None) == 'c'
+    ]
+    c_values = [data['value'] for data in c_data]
+    assert c_values == [1, 2]
```

Comment:
Does `map` guarantee an ordering of the mapped nodes? This seems tricky to pull off given that we rely on NetworkX to store the graph.

Reply:
Cyclebane does not have edge order, but the node names contain the index. The real question is: does Sciline make use of the indices, to guarantee an order in the `reduce` operations? Currently it does not, but I think it probably could.

Reply:
How do the node names factor into this? The code here simply iterates through `result.nodes` but doesn't sort in any way.

Reply:
> Can you clarify if you meant "why does this test pass",

Essentially, yes. I would like to know if the implementation of the graph can change in the future and break the test. E.g., by switching to a different storage layout.

Your comment 

> Cyclebane does not have edge order, but the node names contain the index.

led me to think that because the name includes the dim, the nodes are ordered according to the dim. But I don't see how that factors into this test.

Reply:
I think it is probably because NetworkX is using `dict` internally, i.e., insertion order is preserved?

Reply:
I updated this and other tests to avoid this, see last commit.